File: 프로그래머스/0/120907. OX퀴즈/OX퀴즈.cpp
#include <string>
#include <vector>
#include <iostream>

using namespace std;

vector<string> solution(vector<string> quiz) {
    vector<string> answer;
    vector<string> nums;
    string temp = "";
    int test = 0;
    for (int i = 0; i < quiz.size(); i++) {
        for (int j = 0; j <= quiz[i].length(); j++) {
            if (quiz[i][j] == ' ' || j == quiz[i].length()) {
                if (temp != "=") nums.emplace_back(temp);
                temp = "";
                continue;
            }
            temp += quiz[i][j];
        }
        if(nums[1]=="-") test = stoi(nums[0]) - stoi(nums[2]);
        else test = stoi(nums[0]) + stoi(nums[2]);
        if (test == stoi(nums[3])) answer.emplace_back("O");
        else answer.emplace_back("X");
        nums.clear();
    }
    
    
    return answer;
}

File: 프로그래머스/0/120907. OX퀴즈/OX퀴즈.cs
using System;
using System.Collections.Generic;
using System.Linq;

public class Solution {
    public string[] solution(string[] quiz) {
        string[] answer = new string[] {};
        List<string> answerList = new List<string> {};
        //공백을 기준으로 문자열을 잘라 리스트에 넣는다.
        foreach(string items in quiz){
            List<string> item = items.Split(' ').ToList();
            //num1 + num2값이 result 값과 같은지 판별한다.
            int num1 = int.Parse(item[0]);
            int num2 = int.Parse(item[2]);
            num2 = item[1] == "-" ? -1 * num2 : num2;
            int result = int.Parse(item[4]);
            answerList.Add(num1+num2 == result ? "O" : "X");
        }
        return answer = answerList.ToArray();
    }
}